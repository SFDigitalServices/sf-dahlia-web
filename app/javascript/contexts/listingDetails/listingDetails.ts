interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
