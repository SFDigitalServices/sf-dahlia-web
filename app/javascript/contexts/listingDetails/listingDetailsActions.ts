import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
