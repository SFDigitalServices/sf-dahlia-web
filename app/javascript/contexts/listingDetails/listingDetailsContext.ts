import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
