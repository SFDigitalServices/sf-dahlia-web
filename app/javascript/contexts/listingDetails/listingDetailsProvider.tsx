import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
