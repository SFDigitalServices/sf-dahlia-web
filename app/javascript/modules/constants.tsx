export const PREFERENCES = {
  aliceGriffith: "Alice Griffith Housing Development Resident",
  antiDisplacement: "Anti-Displacement Housing Preference (ADHP)",
  assistedHousing: "Rent Burdened / Assisted Housing Preference",
  rentBurden: "Rent Burdened / Assisted Housing Preference",
  certificateOfPreference: "Certificate of Preference (COP)",
  dalpEducators: "DALP SFUSD Educators",
  dalpFirstResponders: "DALP First Responders",
  dalpGeneral: "DALP General",
  displacedTenant: "Displaced Tenant Housing Preference (DTHP)",
  employmentOrDisability: "Employment or Disability Preference",
  generalLottery: "generalLottery",
  hud221d3: "HUD 221(d)(3) Statutory Preference; Federal or Presidential Disasters (HUD 221)",
  liveWorkInSf: "Live or Work in San Francisco Preference",
  neighborhoodResidence: "Neighborhood Resident Housing Preference (NRHP)",
  rightToReturnSunnydale: "Right to Return - Sunnydale",
  rightToReturnHuntersView: "Right to Return - Hunters View",
  rightToReturnPotrero: "Right to Return - Potrero",
  treasureIsland: "Treasure Island Resident (TIR) Preference",
  veteran: "Veteran",
}

export const PREFERENCES_IDS = {
  rightToReturn: "hope-sf",
  neighborhoodResidence: "nrhp",
  assistedHousing: "rent-burdened-assisted-housing",
  liveWorkInSf: "live-or-work-sf",
  displacedTenant: "dthp",
  certificateOfPreference: "cop",
}

export const PREFERENCES_WITH_PROOF = [
  PREFERENCES.aliceGriffith,
  PREFERENCES.antiDisplacement,
  PREFERENCES.assistedHousing,
  PREFERENCES.liveWorkInSf,
  PREFERENCES.neighborhoodResidence,
  PREFERENCES.rightToReturnSunnydale,
  PREFERENCES.rightToReturnHuntersView,
  PREFERENCES.rightToReturnPotrero,
]

/**
 * 'Reserved Community Type' used at the development or listing level. Not to be
 * confused with the 'Reserved Type' at the unit level
 */
export const RESERVED_COMMUNITY_TYPES = {
  ACCESSIBLE_ONLY: "Accessible Units Only",
  ARTIST: "Artist Live/Work",
  HABITAT: "Habitat for Humanity",
  SENIOR: "Senior",
  VETERAN: "Veteran",
}

export const LISTING_TYPES = {
  OWNERSHIP: "Ownership",
}

export const LISTING_STATUS_ACTIVE = "Active"

export const CUSTOM_LISTING_TYPES = {
  EDUCATOR_ONE: "Educator 1: SFUSD employees only",
  EDUCATOR_TWO: "Educator 2: SFUSD employees & public",
  EDUCATOR_THREE: "Educator 3: Waitlist - SFUSD employees & public",
  DALP: "Downpayment Assistance Loan Program",
}

export const LISTING_TYPE_FIRST_COME_FIRST_SERVED = "First Come, First Served"
export const LISTING_TYPE_STANDARD_LOTTERY = "Standard Lottery"

export const TENURE_TYPES = {
  NEW_SALE: "New sale",
  RESALE: "Resale",
  NEW_RENTAL: "New rental",
  RE_RENTAL: "Re-rental",
}

export const LOTTERY_RANKING_VIDEO_URL = "https://www.youtube.com/watch?v=oW56bUsrSW4"

// Whitelist of allowed SF.gov links for secure redirection
export const SFGOV_LINKS = [
  "https://sf.gov/departments/mayors-office-housing-and-community-development",
  "https://sf.gov/information/learn-about-housing-lottery-preference-programs",
  "https://www.sf.gov/learn-how-lottery-works-shirley-chisholm-village",
  "https://sf.gov/apply-shirley-chisholm-village-housing",
  "https://sf.gov/reports/january-2023/first-come-first-served-bmr-listings",
  "https://sf.gov/reports/december-2022/city-second-program-current-listings",
  "https://sf.gov/information/san-francisco-rental-opportunities",
  "https://sf.gov/reports/august-2023/first-come-first-served-bmr-listings",
  "https://sf.gov/reports/august-2023/city-second-program-current-listings",
  "https://sf.gov/learn-about-certificate-preference-cop",
  "https://sf.gov/displaced-tenant-housing-preference-program-dthp",
  "https://sf.gov/reports/february-2023/find-lender-below-market-rate-program",
  "https://sf.gov/departments/311-customer-service-center",
  "https://sf.gov/sign-complete-homebuyer-education",
  "https://sf.gov/determine-if-you-can-buy-affordable-housing-program",
  "https://sf.gov/reports/october-2023/find-lender-below-market-rate-program",
  "https://sf.gov/departments/city-administrator/digital-services",
  "https://sf.gov/departments/mayors-office-innovation",
  "https://sf.gov/information/special-calculations-household-income",
  "https://sf.gov/after-rental-housing-lottery",
  "https://www.sf.gov/step-by-step/buy-home-without-entering-lottery",
  "https://www.sf.gov/step-by-step--buy-home-without-entering-lottery",
  "https://www.sf.gov/resource/2022/homebuyer-program-counseling-agencies",
  "https://www.sf.gov/dalp-lottery-results",
  // Salesforce links that redirect to sf.gov pages
  "https://www.sf.gov/learn-about-certificate-preference-cop",
  "https://www.sf.gov/get-priority-housing-lottery",
  "https://www.sf.gov/learn-about-displaced-tenant-housing-preference-dthp",
  "https://www.sf.gov/get-priority-housing-lottery",
]

export const DIRECTORY_SECTION_OPEN_LOTTERIES = "open"
export const DIRECTORY_SECTION_FCFS_LISTINGS = "fcfs"
export const DIRECTORY_SECTION_UPCOMING_LOTTERIES = "upcoming"
export const DIRECTORY_SECTION_LOTTERY_RESULTS = "results"
export const DIRECTORY_SECTION_ADDITIONAL_LISTINGS = "additional"

export const SALE_DIRECTORY_SECTIONS = [
  DIRECTORY_SECTION_OPEN_LOTTERIES,
  DIRECTORY_SECTION_FCFS_LISTINGS,
  DIRECTORY_SECTION_UPCOMING_LOTTERIES,
  DIRECTORY_SECTION_LOTTERY_RESULTS,
]
export const RENTAL_DIRECTORY_SECTIONS = [
  DIRECTORY_SECTION_OPEN_LOTTERIES,
  DIRECTORY_SECTION_UPCOMING_LOTTERIES,
  DIRECTORY_SECTION_LOTTERY_RESULTS,
]

export const DIRECTORY_TYPE_SALES = "forSale"
export const DIRECTORY_SECTION_INFO = {
  open: {
    ref: "enter-a-lottery",
    icon: "house",
  },
  fcfs: {
    ref: "buy-now",
  },
  additional: {
    ref: "additional-listings",
    icon: "doubleHouse",
  },
  upcoming: {
    ref: "upcoming-lotteries",
    icon: "clock",
  },
  results: {
    ref: "lottery-results",
    icon: "result",
  },
}

export const DIRECTORY_PAGE_HEADER = "page-header"

// Unleash feature flags
export const UNLEASH_FLAG = {
  CLERK_AUTH: "temp.webapp.auth.clerk",
  FORM_ENGINE: "perm.webapp.formEngine",
  FORM_ENGINE_DEBUG: "perm.webapp.formEngine.debug",
}

export const LISTING_APPLY_FORMS_INPUT_MAX_LENGTH = {
  firstName: 40,
  middleName: 20,
  lastName: 40,
  email: 50,
  password: 50,
  address: 75,
  city: 75,
  alternateContactTypeOther: 50,
  alternateContactAgency: 50,
  certificateNumber: 50,
  genderOther: 50,
  otherLanguage: 50,
  sexualOrientationOther: 50,
  raceEthnicityOther: 50,
}

export const LATIN_REGEX = new RegExp("^[A-z0-9\u00C0-\u017E\\s'.,-/+#%$:=-_`~()]+$")
