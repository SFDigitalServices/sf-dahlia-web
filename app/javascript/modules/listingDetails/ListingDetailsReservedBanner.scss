.educator1-banner {
  @apply normal-case;
  @apply space-y-1;
  @apply tracking-tight;
  font-size: var(--bloom-font-size-xs);
}
