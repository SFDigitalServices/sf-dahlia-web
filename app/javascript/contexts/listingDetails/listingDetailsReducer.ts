import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
