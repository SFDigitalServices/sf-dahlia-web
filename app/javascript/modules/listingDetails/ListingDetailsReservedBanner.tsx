import React from "react"
import { Message, t } from "@bloom-housing/ui-components"
import { RESERVED_COMMUNITY_TYPES } from "../constants"
import { getCustomListingType, renderInlineMarkup } from "../../util/languageUtil"

export interface ListingDetailsReservedBannerProps {
  reservedCommunityMinimumAge?: number
  reservedCommunityType?: string
  customListingType?: string
}

const buildMessage = (
  reservedCommunityMinimumAge?: number,
  reservedCommunityType?: string,
  customListingType?: string
) => {
  if (customListingType && getCustomListingType(customListingType)) {
    return (
      <Message warning={true}>
        {renderInlineMarkup(`${t("listings.customListingType.educator.banner")}`, "<p><b>")}
        {renderInlineMarkup(
          t(`listings.customListingType.educator.banner.link.content`, {
            url: `#listing-details-eligibility`,
          }),
          "<p><a>"
        )}
      </Message>
    )
  }

  if (
    !reservedCommunityType ||
    reservedCommunityType === RESERVED_COMMUNITY_TYPES.HABITAT ||
    !Object.values(RESERVED_COMMUNITY_TYPES).includes(reservedCommunityType)
  )
    return null

  const content =
    reservedCommunityType === RESERVED_COMMUNITY_TYPES.SENIOR
      ? t(`listings.allUnitsReservedFor.${reservedCommunityType}`, {
          age: reservedCommunityMinimumAge,
        })
      : t(`listings.allUnitsReservedFor.${reservedCommunityType}`)

  return <Message warning={true}>{content}</Message>
}

export const ListingDetailsReservedBanner = ({
  reservedCommunityMinimumAge,
  reservedCommunityType,
  customListingType,
}: ListingDetailsReservedBannerProps) => {
  const message = buildMessage(
    reservedCommunityMinimumAge,
    reservedCommunityType,
    customListingType
  )
  if (!message) return
  return <div className="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0">{message}</div>
}
