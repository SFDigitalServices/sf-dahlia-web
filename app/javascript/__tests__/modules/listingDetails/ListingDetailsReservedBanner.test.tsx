import React from "react"
import renderer from "react-test-renderer"
import { ListingDetailsReservedBanner } from "../../../modules/listingDetails/ListingDetailsReservedBanner"
import { RESERVED_COMMUNITY_TYPES } from "../../../modules/constants"

describe("ListingDetailsReservedBanner", () => {
  it("does not display banner when reserved community type is blank", () => {
    const tree = renderer.create(<ListingDetailsReservedBanner />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does not display banner when unknown reserved community type", () => {
    const tree = renderer
      .create(<ListingDetailsReservedBanner reservedCommunityType={"anUnknownType"} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("displays banner when reserved community type is Accessible Units Only", () => {
    const tree = renderer
      .create(
        <ListingDetailsReservedBanner
          reservedCommunityType={RESERVED_COMMUNITY_TYPES.ACCESSIBLE_ONLY}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("displays banner when reserved community type is Artist Live/Work", () => {
    const tree = renderer
      .create(
        <ListingDetailsReservedBanner reservedCommunityType={RESERVED_COMMUNITY_TYPES.ARTIST} />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("does not display banner when reserved community type is Habitat for Humanity", () => {
    const tree = renderer
      .create(
        <ListingDetailsReservedBanner reservedCommunityType={RESERVED_COMMUNITY_TYPES.HABITAT} />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("displays banner when reserved community type is Senior", () => {
    const tree = renderer
      .create(
        <ListingDetailsReservedBanner
          reservedCommunityType={RESERVED_COMMUNITY_TYPES.SENIOR}
          reservedCommunityMinimumAge={65}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("displays banner when reserved community type is Veteran", () => {
    const tree = renderer
      .create(
        <ListingDetailsReservedBanner reservedCommunityType={RESERVED_COMMUNITY_TYPES.VETERAN} />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
