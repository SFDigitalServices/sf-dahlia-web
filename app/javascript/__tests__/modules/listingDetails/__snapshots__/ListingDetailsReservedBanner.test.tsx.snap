// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsReservedBanner displays banner when custom listing type is Educator One 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      <p
        style="text-transform: none;"
      >
        <b>
          Special housing for San Francisco public school employees.
        </b>
      </p>
      <p
        style="text-transform: none;"
      >
        <a
          href="#listing-details-eligibility"
          target="_self"
        >
          Learn more about who can apply lender on our list.
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner displays banner when reserved community type is Accessible Units Only 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      All units reserved for people who need accessibility features
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner displays banner when reserved community type is Artist Live/Work 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      All units reserved for artists to live and work in
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner displays banner when reserved community type is Senior 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      All units reserved for ages 65+
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner displays banner when reserved community type is Veteran 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      All units reserved for veterans
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner displays custom listing type banner when both custom listing and reserved community types are selected 1`] = `
<DocumentFragment>
  <div
    class="md:pr-8 md:w-2/3 mt-4 w-full mb-8 md:mb-0 md:pl-4 lg:pl-0"
  >
    <div
      class="message is-warning"
    >
      <p
        style="text-transform: none;"
      >
        <b>
          Special housing for San Francisco public school employees.
        </b>
      </p>
      <p
        style="text-transform: none;"
      >
        <a
          href="#listing-details-eligibility"
          target="_self"
        >
          Learn more about who can apply lender on our list.
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsReservedBanner does not display banner when custom listing type is unknown 1`] = `<DocumentFragment />`;

exports[`ListingDetailsReservedBanner does not display banner when reserved community and custom listing types are blank 1`] = `<DocumentFragment />`;

exports[`ListingDetailsReservedBanner does not display banner when reserved community type is Habitat for Humanity 1`] = `<DocumentFragment />`;

exports[`ListingDetailsReservedBanner does not display banner when unknown reserved community type 1`] = `<DocumentFragment />`;
